types/bluebird/index.d.ts

```diff
@@ -490,11 +490,11 @@ declare class Bluebird<R> implements Bluebird.Thenable<R>, Bluebird.Inspection<R
    *
    * *The original object is not modified.*
    */
-  // TODO verify this is correct
+  // TODO disallow non-objects
   // trusted promise for object
-  static props(object: Bluebird<Object>): Bluebird<Object>;
+  static props<T>(object: Bluebird.Thenable<{ [K in keyof T]: Bluebird.Thenable<T[K]> | T[K] }>): Bluebird<{ [K in keyof T]: T[K] }>;
```

Comment:
Would it be useful to create a type alias for `{ [K in keyof T]: Bluebird.Thenable<T[K]> | T[K] }>` like `ThenableProps<T>`?